tests/test_benchmarking/test_caching/test_simulation_cache.py

```diff
@@ -0,0 +1,256 @@
+import json
+import os
+import shutil
+import unittest
+from os import path
+from typing import List
+
+from api.models.student import Student
+from api.models.team import Team
+from api.models.team_set import TeamSet
+from api.models.team_set.serializer import TeamSetSerializer
+from benchmarking.caching.simulation_cache import SimulationCache
+from benchmarking.simulation.simulation import SimulationArtifact
+
+mock_simulation_result: List[TeamSet] = [
+    TeamSet(
+        _id=0,
+        name="TeamSet0",
+        teams=[
+            Team(
+                _id=0,
+                name="Team0",
+                students=[
+                    Student(_id=0),
+                    Student(_id=1),
+                ],
+            ),
+            Team(
+                _id=1,
+                name="Team1",
+                students=[
+                    Student(_id=2),
+                    Student(_id=3),
+                ],
+            ),
+        ],
+    ),
+    TeamSet(
+        _id=1,
+        name="TeamSet1",
+        teams=[
+            Team(
+                _id=0,
+                name="Team0",
+                students=[
+                    Student(_id=0),
+                    Student(_id=2),
+                ],
+            ),
+            Team(
+                _id=1,
+                name="Team1",
+                students=[
+                    Student(_id=1),
+                    Student(_id=3),
+                ],
+            ),
+        ],
+    ),
+]
+mock_runtimes: List[float] = [
+    1.8,
+    2.0,
+]
+
+
+class TestSimulationCache(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # I'm paranoid about these tests modifying the cache, so move the cache to a different location as a precaution
```

Comment:
something you also might be able to do is just mock a bunch of the `os` methods and have it "make files" and "open files" from somewhere else instead. 

it's a bit sus to do a full file copy tbh 😭 so I think we should explore an approach that uses mocks to still test core functionality 

i.e. if you mock the response of `path.dirname(__file__)`, you can make it think you are anywhere you want in the file structure

In other cases, you can just mock a call to the os function that deletes files and just make sure it was called, instead of checking that an actual file was actually deleted.

You could also create a `prepare_cache_data` method in SimulationCache that returns the exact json that will be saved, so you can unit test that fields are in that dict and have the only thing that actually needs to check if it created a file be the single "save_data" method or whatever

I'm trying to drop ideas but feel free to call me to chat about this - I think doing shenanigan directory moving is a scary thing

Reply:
Took me a while to figure out how to mock stuff, but it works well